atomate/vasp/workflows/base/magnetism.py

```diff
@@ -0,0 +1,871 @@
+            object for later database ingestion.
+
+            Args:
+                structure: Structure
+
+            Returns: TransformedStructure
+            """
+            # this could be further improved by storing full transformation
+            # history, but would require an improved transformation pipeline
+            return TransformedStructure(
+                structure, other_parameters={"wf_meta": self.wf_meta}
+            )
+
+        ordered_structures = [_add_metadata(struct) for struct in ordered_structures]
+
+        # in case we've introduced duplicates, let's remove them
+        logger.info("Pruning duplicate structures.")
+        structures_to_remove = []
+        for idx, ordered_structure in enumerate(ordered_structures):
+            if idx not in structures_to_remove:
+                duplicate_checker = CollinearMagneticStructureAnalyzer(
+                    ordered_structure, overwrite_magmom_mode="none"
+                )
+                for check_idx, check_structure in enumerate(ordered_structures):
+                    if check_idx not in structures_to_remove and check_idx != idx:
+                        if duplicate_checker.matches_ordering(check_structure):
+                            structures_to_remove.append(check_idx)
+
+        if len(structures_to_remove):
+            logger.info(
+                "Removing {} duplicate ordered structures".format(
+                    len(structures_to_remove)
+                )
+            )
+            ordered_structures = [
+                s
+                for idx, s in enumerate(ordered_structures)
+                if idx not in structures_to_remove
+            ]
+            ordered_structures_origins = [
+                o
+                for idx, o in enumerate(ordered_structures_origins)
+                if idx not in structures_to_remove
+            ]
+
+        # also remove low symmetry structures
+        if self.truncate_by_symmetry:
+
+            # by default, keep structures with 5 most symmetric space groups
+            if not isinstance(self.truncate_by_symmetry, int):
+                self.truncate_by_symmetry = 5
+
+            logger.info("Pruning low symmetry structures.")
+
+            # first get a list of symmetries present
+            symmetry_int_numbers = [
+                s.get_space_group_info()[1] for s in ordered_structures
+            ]
+
+            # then count the number of symmetry operations for that space group
+            num_sym_ops = [
+                len(SpaceGroup.from_int_number(n).symmetry_ops)
+                for n in symmetry_int_numbers
+            ]
+
+            # find the largest values...
+            max_symmetries = sorted(list(set(num_sym_ops)), reverse=True)
+
+            # ...and decide which ones to keep
+            if len(max_symmetries) > self.truncate_by_symmetry:
+                max_symmetries = max_symmetries[0:5]
```

Comment:
why is this ```0:5``` and not ```0:self.truncate_by_symmetry```

Reply:
Thanks for catching this!

Reply:
This is consistent with the docstring, but agreed would make more sense to be able to set this arbitrarily.